### PR TITLE
Enable cluster installation from local sources

### PR DESCRIPTION
**Pull request by suleymanakbas91**
**Description**

Changes proposed in this pull request:

- Enable cluster installation from local sources
- The command looks like:
```bash
kyma install -s local --docker-username <username> --docker-password <password> --custom-image <image-registry>/<image-name>:<tag>
```

**Related issue(s)**

See https://github.com/kyma-project/cli/issues/334

### CODE REVIEW

File: pkg/installation/utils.go [hunk 1]
```diff
@@ -63,6 +77,54 @@ func (i *Installation) buildKymaInstaller(imageName string) error {
 	return nil
 }
 
+func (i *Installation) pushKymaInstaller(imageName string) error {
```
Comment (clebs) [inline pkg/installation/utils.go:80]: Why is the image name passed as parameter if we already have it in `i.Options.CustomImage`?

Comment (suleymanakbas91) [reply pkg/installation/utils.go:80]: I just wanted to keep it as similar as possible to buildKymaInstaller function. There is no other reason. :)

Comment (suleymanakbas91) [reply pkg/installation/utils.go]: I changed it

File: cmd/kyma/install/cmd.go [hunk 2]
```diff
@@ -84,15 +84,18 @@ The standard installation uses the minimal configuration. The system performs th
...
 	cobraCmd.Flags().StringVarP(&o.Password, "password", "p", "", "Predefined cluster password.")
 	cobraCmd.Flags().StringArrayVarP(&o.OverrideConfigs, "override", "o", nil, "Path to a YAML file with parameters to override.")
 	cobraCmd.Flags().StringVarP(&o.ComponentsConfig, "components", "c", "", "Path to a YAML file with component list to override.")
 	cobraCmd.Flags().IntVar(&o.FallbackLevel, "fallbackLevel", 5, `If "source=latest-published", defines the number of commits from master branch taken into account if artifacts for newer commits do not exist yet`)
+	cobraCmd.Flags().StringVarP(&o.DockerUsername, "docker-username", "", "", "Docker username to push the custom image. Only used for installation from local sources to a remote cluster.")
+	cobraCmd.Flags().StringVarP(&o.DockerPassword, "docker-password", "", "", "Docker password to push the custom image. Only used for installation from local sources to a remote cluster.")
+	cobraCmd.Flags().StringVarP(&o.CustomImage, "custom-image", "", "", "Full image name including the registry and the tag. Only used for installation from local sources to a remote cluster.")
```
Comment (bszwarc) [inline cmd/kyma/install/cmd.go:98]:
```suggestion
	cobraCmd.Flags().StringVarP(&o.DockerUsername, "docker-username", "", "", "Docker username to push the custom image. Used only for installation from local sources to a remote cluster.")
	cobraCmd.Flags().StringVarP(&o.DockerPassword, "docker-password", "", "", "Docker password to push the custom image. Used only for installation from local sources to a remote cluster.")
	cobraCmd.Flags().StringVarP(&o.CustomImage, "custom-image", "", "", "Full image name including the registry and the tag. Used only for installation from local sources to a remote cluster.")
```

File: docs/gen-docs/kyma_install.md [hunk 3]
```diff
@@ -45,23 +45,26 @@ kyma install [flags]
...
-                               	- To use the local sources, write "kyma install --source=local". 
-                               	- To use a custom installer image, write kyma "install --source=user/my-kyma-installer:v1.4.0".
-      --src-path string        Absolute path to local sources.
-      --timeout duration       Time-out after which CLI stops watching the installation progress. (default 1h0m0s)
-      --tlsCert string         TLS certificate for the domain used for installation. The certificate must be a base64-encoded value.
-      --tlsKey string          TLS key for the domain used for installation. The key must be a base64-encoded value.
+  -c, --components string        Path to a YAML file with component list to override.
+      --custom-image string      Full image name including the registry and the tag. Only used for installation from local sources to a remote cluster.
+      --docker-password string   Docker password to push the custom image. Only used for installation from local sources to a remote cluster.
+      --docker-username string   Docker username to push the custom image. Only used for installation from local sources to a remote cluster.
```
Comment (bszwarc) [inline docs/gen-docs/kyma_install.md:51]:
```suggestion
      --custom-image string      Full image name including the registry and the tag. Used only for installation from local sources to a remote cluster.
      --docker-password string   Docker password to push the custom image. Used only for installation from local sources to a remote cluster.
      --docker-username string   Docker username to push the custom image. Used only for installation from local sources to a remote cluster.
```

**Pull request merged**